general/utility/general_graphics/source/com/robin/general/graphics/ForceVector.java

```diff
@@ -3,105 +3,210 @@
+        double fx = dx + fv.dx;
+        double fy = dy + fv.dy;
+
+        // Now we simply need to identify the new force and direction
+        double fForce = Math.sqrt((fx * fx) + (fy * fy));
+        double fDirection = toDegrees(Math.atan2(fy, fx));
+
+        return new ForceVector(fForce, fDirection, fx, fy);
+    }
+
+    // makes sure the angle is between 0 and 360
+    private void normalizeAngle() {
+        while (direction < DEGREES_0) {
+            direction += DEGREES_360;
+        }
+        while (direction >= DEGREES_360) {
+            direction -= DEGREES_360;
+        }
+    }
+
+    // compute x,y coordinates of vector (assuming 0,0 origin)
+    private void initRectangularCoordinates() {
+        normalizeAngle();
+        double radians = toRadians(direction);
+        dx = force * Math.cos(radians);
+        dy = force * Math.sin(radians);
+    }
+
+
+    @Override
+    public String toString() {
+        return "p(" + NF_2DP.format(force) + "," + NF_2DP.format(direction) +
+                "o)==r(" + NF_2DP.format(dx) + "," + NF_2DP.format(dy) + ")";
+    }
+
+    /**
+     * Returns the force (magnitude) of this force vector.
+     *
+     * @return the force
+     */
+    public double getForce() {
+        return force;
+    }
+
+    /**
+     * Returns the direction of this force vector, in degrees (0-360).
+     *
+     * @return the direction
+     */
+    public double getDirection() {
+        return direction;
+    }
+
+    /**
+     * Returns the x component of this force vector.
+     *
+     * @return the x component
+     */
+    public double getDx() {
+        return dx;
+    }
+
+    /**
+     * Returns the y component of this force vector.
+     *
+     * @return the y component
+     */
+    public double getDy() {
+        return dy;
+    }
+
+    /**
+     * Converts the given radians to degrees.
+     *
+     * @param rad radians
+     * @return value in degrees
+     */
+    public static double toDegrees(double rad) {
+        // The minus sign flips the angle so
+        // that 90 degrees is pointing straight up
+        return -(rad * 180) / Math.PI;
+    }
+
+    /**
+     * Converts the given degrees to radians.
+     *
+     * @param deg degrees
+     * @return value in radians
+     */
+    public static double toRadians(double deg) {
+        // The minus sign flips the angle so
+        // that 90 degrees is pointing straight up
+        return -(Math.PI * deg) / 180.0;
+    }
+
+
+    // TODO: remove, in deference to unit tests
+    @Deprecated
+    public static void main(String[] args) {
+        System.out.println(new ForceVector(1.0, 0.0));
+        System.out.println(new ForceVector(1.0, 90.0));
+        System.out.println(new ForceVector(1.0, 180.0));
+        System.out.println(new ForceVector(1.0, 270.0));
+
+        ForceVector fv1 = new ForceVector(1.0, 0.0);
+        ForceVector fv2 = new ForceVector(1.0, 90.0);
+        System.out.println(fv1.add(fv2));
+    }
```

Comment:
Did the spacing change here?  Tabs to spaces?

BTW - ForceVector isn't used anywhere in RealmSpeak.  I think this is a utility class I had in here for something else, and it just came along for the ride.  Still, I appreciate you making an effort to add some code coverage!